src/figuresClasses.ts

```diff
@@ -2,10 +1,95 @@
+// import { error } from 'console';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: `red` | `green` | `blue`;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: 'triangle';
+
+  color: 'red' | 'green' | 'blue';
+
+  private a: number;
+
+  private b: number;
+
+  private c: number;
+
+  constructor(
+    color: 'red' | 'green' | 'blue',
+    a: number,
+    b: number,
+    c: number,
+  ) {
+    if (a >= b + c || b >= a + c || c >= b + a) {
+      throw new Error(`Sides 1, 2 and 3 can't form a triangle.`);
+    }
+    this.shape = 'triangle';
+    this.color = color;
+    this.a = a;
+    this.b = b;
+    this.c = c;
+  }
+
+  getArea(): number {
+    const semiP = 0.5 * (this.a + this.b + this.c);
+    const square = Math.sqrt(
+      semiP * (semiP - this.a) * (semiP - this.b) * (semiP - this.c),
+    );
+
+    return Number(square.toFixed(2));
+  }
+}
+
+export class Circle implements Figure {
+  shape: 'circle';
```

Comment:
The `shape` property should be declared as a class property with a type annotation, similar to `color`. It should be `shape: 'circle' = 'circle';` to properly initialize and type the property.